presto-hive/src/main/java/com/facebook/presto/hive/HivePageSink.java

```diff
@@ -283,7 +284,7 @@ private HiveRecordWriter createWriter(List<Object> partitionRow)
         checkArgument(partitionRow.size() == partitionColumnNames.size(), "size of partitionRow is different from partitionColumnNames");
 
         List<String> partitionValues = partitionRow.stream()
-                .map(Object::toString) // todo this seems wrong
+                .map(Objects::toString)
```

Comment:
I hope I understand what you said correctly @electrum. If I am not missing anything here, the new implementation seems good, let me know If should push it. 

See results below:

Here is my test hive table:
~~~
CREATE TABLE null_test_string (
    x varchar, 
    y varchar
) 
WITH (partitioned_by = ARRAY['y']);
~~~

After I insert data with following query:
~~~
insert into table null_test_string partition(y)
select * from 
(
select stack(
    2,
    'hive1', NULL,
    'hive2', NULL
    ) as (x, y) 
) s;
~~~
and inserting via Presto after I modified the code as @yuananf suggested.
~~~
INSERT INTO nulltest (x) VALUES ('0.131');
~~~
`select *` gives me results like this:
~~~
hive1	__HIVE_DEFAULT_PARTITION__
hive2	__HIVE_DEFAULT_PARTITION__
0.131	__HIVE_DEFAULT_PARTITION__
~~~
I did same test for the case where partition key is `bigint`
~~~
biginttest1	NULL
biginttest2	NULL
0.131	NULL
~~~
As you can see above row starting with 0.131 inserted by Presto, others inserted by Hive. Seems like Presto handles this case correctly.

For other types:

timestamp:
~~~
timestamptest1	NULL
timestamptest2	NULL
0.131	NULL
~~~

binary:
~~~
binarytest1	__HIVE_DEFAULT_PARTITION__
binarytest2	__HIVE_DEFAULT_PARTITION__
0.131	__HIVE_DEFAULT_PARTITION__
~~~

date:
~~~
datetest1	NULL
datetest2	NULL
0.131	NULL
~~~

double:
~~~
doubletest1	NULL
doubletest2	NULL
0.131	NULL
~~~

boolean:
~~~
booleantest1	true
booleantest2	true
0.131	true
~~~
FYI: For some of them I inserted data from another table without using stack.